frontend/src/components/worksheets/Worksheet/Worksheet.js

```diff
@@ -997,7 +990,7 @@ class Worksheet extends React.Component {
                 }
                 var rawIndex = editor.getCursorPosition().row;
                 this.setState({
-                    editMode: editMode,
+                    inSourceEditMode: openSourceEditMode,
```

Comment:
To make the code even clearer, can we explicitly set this

```suggestion
                    inSourceEditMode: false,
```